### ISSUE TITLE
Broken path to example with strpdate2num

### ISSUE THREAD

**Issue by kiddick** [matplotlib/matplotlib#7437]
Seems it is outdated path to example:
 :file:`examples/load_demo.py`.

https://github.com/matplotlib/matplotlib/blob/5ac0881f770c4cb0f67d0625b9762264a3cbdfa2/lib/matplotlib/dates.py#L278

**Comment by vraelvrangr** [comment]
So I looked at the file you pointed out and saw that actually the class bytespdate2num also has a broken link to an example case, too. Searching the repository for any reference to strpdate2num does not result in any example use cases. bytespdate2num however does have an example use case:

https://github.com/matplotlib/matplotlib/blob/38be7aeaaac3691560aeadafe46722dda427ef47/examples/pylab_examples/load_converter.py 

It seems that either function should do the same thing so I tried to simply make essentially a copy of the file using strpdate2num instead of bytespdate2num. It works the same when run in Python2 but in Python3 it seems that the __call__ function in strpdate2num receives the parameter s as a bytes type rather than string. Personally I am not sure how I might go about resolving that, but I suppose another example case file could be made. I am interested into why an error might be showing up though when I try to simply replace bytepdatetonum with strpdate2num in the example case.
